job/server/src/main/java/alluxio/worker/JobWorker.java

```diff
@@ -69,8 +68,7 @@
    * @param ufsManager the ufs manager
    */
   JobWorker(FileSystem filesystem, FileSystemContext fsContext, UfsManager ufsManager) {
-    super(
-        Executors.newFixedThreadPool(1, ThreadFactoryUtils.build("job-worker-heartbeat-%d", true)));
+    super(ExecutorServiceFactories.fixedThreadPool("job-worker-executor", 1));
```

Comment:
is the thread rename intentional?

Reply:
Yes it's actually an worker executor thread. Various heartbeats are running on them.